marlowe-runtime/cip30-demo/Main.hs

```diff
@@ -21,9 +37,22 @@ newtype WalletName = WalletName String
     , Blaze.ToMarkup
     )
 
+newtype PostWalletCreateRequestDTO = PostWalletCreateRequestDTO { source :: Text }
```

Comment:
You'll need more information besides the contract source, and the contract can be a stronger type than this (assuming we `JSON.parse` on the frontend before adding it to the `fetch` request). Check out `PostContractsRequest` in `L.M.R.Web.Types` to see what is needed. Additionally, we need to include the change address, extra addresses, and collateral UTxOs from the CIP-30 wallet in this request. The web server passes these via headers, but you can just put them in the request body here.

Reply:
I absolutely agree, the `createContract` procedure will force me to provide more data and I agree that it can be cleaned up and architectured better.